controlpanel/frontend/views/user.py

```diff
@@ -6,12 +7,37 @@
 from django.views.generic.edit import DeleteView, UpdateView
 from django.views.generic.list import ListView
 
+from controlpanel.api import auth0
 from controlpanel.api.models import User
 
 
 class UserList(LoginRequiredMixin, ListView):
     model = User
     template_name = "user-list.html"
+    queryset = User.objects.exclude(auth0_id='')
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+
+        users = auth0.ManagementAPI().list_users()
+
+        last_logins = []
+        for user in users:
+            auth0_id = user['user_id']
+
+            if auth0_id == '':
+                continue
+
+            try:
+                last_login = dateutil.parser.parse(user.get('last_login'))
+            except (TypeError, ValueError):
+                last_login = None
+
+            last_logins.append((auth0_id, last_login))
```

Comment:
Python noob question: What's the benefit of appending tuples compared to just have `last_logins` as a dictionary, e.g.

```python
last_logins = {}

for user in users:
   # ...
   last_logins[auth0_id] = last_login
```

or even doing that directly on `context['last_login']`?

Reply:
There's no difference really. I suppose it might be faster to avoid those dictionary lookups when assigning, but I don't know if `dict(list_of_tuples)` will do that anyway. I started with a dict-comprehension, but had to catch the exceptions.